openmp/libomptarget/tools/kernelreplay/llvm-omp-kernel-replay.cpp

```diff
@@ -23,6 +23,9 @@
 
 using namespace llvm;
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wglobal-constructors"
```

Comment:
Does Clang and other non-GCC compilers honor this as well?

Reply:
Clang does warn about unknown pragmas with `-Wall`, but seems to actually support `'#pragma GCC diagnostic`: https://godbolt.org/z/WfTrMGr5d

Reply:
A better solution in the future would probably be to just make this tool not link directly with the plugin object.